src/libstd/sys/windows/mod.rs

```diff
@@ -81,10 +81,55 @@ pub fn decode_error_kind(errno: i32) -> ErrorKind {
     }
 }
 
+// Ported from <https://github.com/bminor/glibc/blob/master/wcsmbs/wmemchr.c>.
```

Comment:
does this have licensing implications?

Reply:
I was thinking about that too. Given that loop unrolling technique is quite
well-known, I think at least we could ask for permission from glibc maintainers.
Even bettern we could have lawyers' consultant.
Maybe there could be better way to handle this situation.

Reply:
cc @skade, @pnkfelix, who have been heading up the licensing discussion thus far

Reply:
How "well-known" is this technique? Are there other examples of it somewhere else or is it documented in some other literature?

Definitely, putting in "this is a straight port from a GCC" has licensing implications.

Reply:
IANAL, but I don't see an issue here. If you want to link to something other than gcc, then you could defensibly link to the [loop unrolling Wikipedia article](https://en.wikipedia.org/wiki/Loop_unrolling), as the code is a straight-forward application of the ideas described there.

Reply:
@BurntSushi That was exactly where I was coming from. If this is a very widely know technique, I would drop the mention.

Reply:
Thanks all. I dropped the mention.